packages/cdktf-cli/bin/cmds/ui/synth.tsx

```diff
@@ -1,52 +1,67 @@
 import React from "react";
-import { Box, Text } from "ink";
+import { Text } from "ink";
 
 import { useCdktfProject } from "./hooks/cdktf-project";
 import { StreamView, StatusBottomBar } from "./components";
-import { SynthesizedStack } from "../../../lib";
+import { CdktfProject, SynthesizedStack } from "../../../lib";
 
 interface CommonSynthConfig {
   outDir: string;
-  targetStack: string;
-  jsonOutput: boolean;
+  targetStacks: string[];
 }
 
 interface SynthConfig extends CommonSynthConfig {
   synthCommand: string;
 }
 type SynthOutputConfig = {
-  jsonOutput: boolean;
-  currentStackName: string;
+  targetStacks: string[];
   stacks: SynthesizedStack[];
 };
-const SynthOutput = ({
-  jsonOutput,
-  currentStackName,
-  stacks,
-}: SynthOutputConfig): React.ReactElement => {
-  const stack =
-    stacks.find((item) => item.name === currentStackName) || stacks[0];
+
+export async function synthToJSON({
```

Comment:
nit:
```suggestion
export async function synthToJson({
```
would be more consistent with our spelling of JSON elsewhere in the codebase 😅